bothan-api/server-cli/src/commands/query.rs

```diff
@@ -0,0 +1,466 @@
+use std::collections::{HashMap, HashSet};
+use std::error::Error as StdError;
+use std::time::Duration;
+
+use anyhow::anyhow;
+use bothan_api::config::AppConfig;
+use bothan_lib::types::AssetInfo;
+use bothan_lib::worker::rest::AssetInfoProvider as RestAssetInfoProvider;
+use bothan_lib::worker::websocket::{AssetInfoProvider as WebSocketAssetInfoProvider, Data};
+use clap::{Parser, Subcommand};
+use futures::stream::{FuturesOrdered, StreamExt};
+use humantime::Duration as HumanDuration;
+use itertools::Itertools;
+use tokio::time::timeout;
+
+#[derive(Parser)]
+pub struct QueryCli {
+    #[command(subcommand)]
+    subcommand: QuerySubCommand,
+}
+
+#[derive(Subcommand)]
+enum QuerySubCommand {
+    /// Query Binance prices
+    Binance {
+        /// The list of query ids to query prices for
+        query_ids: Vec<String>,
+
+        /// Timeout duration
+        #[arg(short, long, default_value = "60s")]
```

Comment:
60s is quite a long default timeout value, i think something shorter would be more appropriate